client/internal/wssender.go

```diff
@@ -25,15 +27,24 @@ type WSSender struct {
 	// Indicates that the sender has fully stopped.
 	stopped chan struct{}
 	err     error
+
+	heartbeatIntervalUpdated chan struct{}
+	heartbeatIntervalSeconds atomic.Int64
+	heartbeatTimer           *time.Timer
 }
 
 // NewSender creates a new Sender that uses WebSocket to send
 // messages to the server.
 func NewSender(logger types.Logger) *WSSender {
-	return &WSSender{
+	s := &WSSender{
 		logger:       logger,
 		SenderCommon: NewSenderCommon(),
 	}
+	s.heartbeatIntervalUpdated = make(chan struct{}, 1)
+	s.heartbeatIntervalSeconds.Store(defaultHeartbeatIntervalSeconds)
+	s.heartbeatTimer = time.NewTimer(0)
```

Comment:
Is there a reason we're going with a timer instead of a ticker? In my custom implementation of [this](https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/operator-opamp-bridge/agent/agent.go#L64), i used a ticker and I think the code is a bit easier to reason about. Though maybe doing the interval reset is more challenging

Reply:
The [specification](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#opampconnectionsettingsheartbeat_interval_seconds) states that:
> The Agent must count heartbeating interval from the last AgentToServer message sent.

I think a timer is preferable to a ticker as the interval should be reset after each AgentToServer message.